core/types/block.go

```diff
@@ -79,7 +79,7 @@ type Header struct {
 	GasLimit    uint64         `json:"gasLimit"         gencodec:"required"`
 	GasUsed     uint64         `json:"gasUsed"          gencodec:"required"`
 	Time        uint64         `json:"timestamp"        gencodec:"required"`
-	Extra       []byte         `json:"extraData"        gencodec:"required"`
+	Extra       []byte         `json:"extraData"        gencodec:"required" rlp:"-"`
```

Comment:
I'm wondering if we ignore this with RLP encoding, do we still send this field when forwarding blocks in the network layer? How do we serialize and deserialize blocks there?

Reply:
Fields with rlp:"-" are [not used during serialization and deserialization](https://pkg.go.dev/github.com/ethereum/go-ethereum/rlp).

Reply:
This is actually an issue now, as this is used by Clique currently. Therefore I cannot do this. Seems you requested this modification, do you have an idea of a workaround that satisfies your requirements? @jonastheis

Reply:
I think the requirements are:
- Include signature (`extraData`) when broadcasting/syncing the block.
- If `systemConfig` consensus is active:
    - Enforce some constraints on `Difficulty` and `ExtraData`. (The latter is exactly 32 bytes i.e. a signature, no vanity tag allowed.)
    - Exclude the signature from the block hash. (--> cannot use the default RLP encoding during block hashing)

Reply:
Actually this was done by Morty, so my new commit needs not introduce new code for the above.